client/js/components/GameView/MoveChoiceOverlay.vue

```diff
@@ -70,166 +70,175 @@ export default {
+      };
+      return oneOffMove;
+    },
 
-      switch (cardRank) {
+    oneOffTwoAndNineMove() {
+      if (!this.selectedCard) return [];
+      let oneOffDisabled = !this.isPlayersTurn;
+      let oneOffDisabledExplanation = "It's not your turn";
+      if (this.isPlayersTurn) {
+        if (this.opponentQueenCount >= 2) {
+          oneOffDisabled = true;
+          oneOffDisabledExplanation = `You can't play a ${this.selectedCard.rank} while your opponent has two or more queens`;
+        } else {
+          let validTargetExists;
+          // Twos
+          if (this.selectedCard.rank === 2) {
+            const numOpFaceCards = this.$store.getters.opponent.faceCards.length;
+            const numOpJacks = this.$store.getters.opponent.points.reduce(
+              (jackCount, pointCard) => {
+                return jackCount + pointCard.attachments.length;
+              },
+              0
+            );
+            const numTotalTargets = numOpFaceCards + numOpJacks;
+            validTargetExists = numTotalTargets >= 1;
+            if (this.allMovesAreDisabled) {
+              oneOffDisabledExplanation = this.disabledText;
+            } else if (!validTargetExists) {
+              oneOffDisabled = true;
+              oneOffDisabledExplanation = 'There are no Royals to target';
+            }
+            // Nines
+          } else {
+            const numValidTargets =
+              this.$store.getters.opponent.points.length +
+              this.$store.getters.opponent.faceCards.length;
+            if (this.allMovesAreDisabled) {
+              oneOffDisabledExplanation = this.disabledText;
+            } else if (numValidTargets === 0) {
+              oneOffDisabled = true;
+              oneOffDisabledExplanation = 'There are no point cards or Royals to target';
+            }
+          }
+        }
+      }
+      const twoAndNine = {
+        displayName: 'One-Off',
+        eventName: 'targetedOneOff',
+        moveDescription: this.selectedCard.ruleText,
+        disabled: oneOffDisabled || this.allMovesAreDisabled,
+        disabledExplanation: oneOffDisabledExplanation,
+      };
+      return twoAndNine;
+    },
+    oneOffJackMove() {
+      let ableToJack = false;
+      let disabledExplanation = '';
+      if (this.isPlayersTurn && !this.allMovesAreDisabled) {
+        ableToJack = this.opponentQueenCount === 0;
+        disabledExplanation = "You cannot jack your opponent's points while they have a queen";
+      } else {
+        disabledExplanation = this.disabledText;
+      }
+      const oneOffJack = {
+        displayName: 'Royal',
+        eventName: 'jack',
+        moveDescription: "Steal an opponent's point card",
+        disabled: !ableToJack || this.allMovesAreDisabled,
+        disabledExplanation,
+      };
+      return oneOffJack;
+    },
+
+    /**
+     * Returns list of objects representing the available moves,
+     * based on the selected card
+     */
+    moveChoices() {
```

Comment:
Let's remove the `res` variable and the `break` statements and just `return` the array that were assigning to `res` in this function